packages/main/src/RPA/Robocorp/Storage.py

```diff
@@ -78,61 +81,229 @@ def print_all_assets():
+        """Creates or updates an asset named `name` with the provided dictionary
+        `value`.
 
         :param name: Name of the existing or new asset to create (if missing)
-        :param value: The new value to set within the asset
-        :param wait: Wait for value to be set successfully
+        :param value: The new dictionary value to set within the asset
+        :param wait: Wait for the value to be set successfully
 
         **Example: Robot Framework**
 
         .. code-block:: robotframework
 
             *** Tasks ***
             Set An Asset
-                Set Asset    my-asset    My string asset value
+                &{entries} =    Create Dictionary   dogs    ${10}
+                Set JSON Asset    my-json-asset    ${entries}
 
         **Example: Python**
 
         .. code-block:: python
 
             def set_an_asset():
-                storage.set_asset("my-asset", "My string asset value")
+                entries = {"dogs": 10}
+                storage.set_json_asset("my-json-asset", entries)
         """
-        storage.set_asset(name, value, wait=wait)
+        storage.set_json(name, value=value, wait=wait)
+
+    @keyword
+    def get_json_asset(self, name: str) -> Dict:
+        """Get the asset's dictionary value by providing its `name`.
+
+        :param name: Name of the asset
+        :raises AssetNotFound: Asset with the given name does not exist
+        :returns: The current value of this asset as a dictionary
+
+        **Example: Robot Framework**
+
+        .. code-block:: robotframework
+
+            *** Tasks ***
+            Retrieve An Asset
+                &{value} =      Get JSON Asset       my-json-asset
+                Log     Asset dictionary value: ${value}
+
+        **Example: Python**
+
+        .. code-block:: python
+
+            def retrieve_an_asset():
+                value = storage.get_json_asset("my-json-asset")
+                print("Asset dictionary value:", value)
+        """
+        return storage.get_json(name)
+
+    @keyword
+    def set_file_asset(self, name: str, path: str, wait: bool = True):
+        """Creates or updates an asset named `name` with the content of the given
+        `path` file.
+
+        :param name: Name of the existing or new asset to create (if missing)
+        :param path: The file path whose content to be set within the asset
+        :param wait: Wait for the value to be set successfully
+
+        **Example: Robot Framework**
+
+        .. code-block:: robotframework
+
+            *** Tasks ***
+            Set An Asset
+                Set File Asset    my-file-asset      report.pdf
+
+        **Example: Python**
+
+        .. code-block:: python
+
+            def set_an_asset():
+                storage.set_file_asset("my-file-asset", "report.pdf")
+        """
+        storage.set_file(name, path=path, wait=wait)
+
+    @keyword
+    def get_file_asset(self, name: str, path: str, overwrite: bool = False) -> str:
```

Comment:
Usually `overwrite` is more appropriate since we aren't just fine if a file exists with `exist_ok`, because the end result isn't the same, as we might end up with a completely different file content even if the name is the same. (`exist_ok` applies best when making a directory and we're just ok if it already exists, as it doesn't produce an undesired effect @osrjv -- no need to change this in **robo** as well)